ext/civi_case/Civi/Afform/Behavior/ContactAutofillBasedOnCase.php

```diff
@@ -0,0 +1,212 @@
+<?php
+namespace Civi\Afform\Behavior;
+
+use Civi\Afform\Event\AfformEntitySortEvent;
+use Civi\Afform\Event\AfformPrefillEvent;
+use Civi\API\Event\RespondEvent;
+use Civi\Api4\Utils\CoreUtil;
+use Civi\Core\Event\GenericHookEvent;
+use Civi\Core\Service\AutoService;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use CRM_Case_ExtensionUtil as E;
+
+/**
+ * @service
+ * @internal
+ */
+class ContactAutofillBasedOnCase extends AutoService implements EventSubscriberInterface {
+
+  /**
+   * @return array
+   */
+  public static function getSubscribedEvents() {
+    return [
+      'civi.api.respond' => 'onAfformGetBehavors',
+      'hook_civicrm_angularModules' => 'onAngularModules',
+      'hook_civicrm_buildAsset' => 'onBuildAsset',
+      'civi.afform.sort.prefill' => 'onAfformSortPrefill',
+      'civi.afform.prefill' => ['onAfformPrefill', 100],
+    ];
+  }
+
+  public static function onAfformGetBehavors(RespondEvent $event) {
+    if ($event->getEntityName() == 'AfformBehavior' && $event->getActionName() == 'get') {
+      $response = $event->getResponse();
+      foreach ($response as $index => $behavior) {
+        foreach (\CRM_Contact_BAO_ContactType::basicTypes() as $contactType) {
```

Comment:
This looks like it would alter every behavior but you only want to target one.
Before this loop you should add a condition `if ($behavior['key'] === 'autofill')`